armi/physics/fuelCycle/hexAssemblyFuelMgmtUtils.py

```diff
@@ -19,108 +19,122 @@
-    bIndexMaxBu = a.index(maxBuBlock)
-
-    if maxBuPinIndexAssem == 0:
-        # Don't bother rotating if the highest-BU pin is the central pin. End this method.
-        return rot
+    maxBuBlock = maxBurnupBlock(a)
+    if maxBuBlock.spatialGrid is None:
+        raise ValueError(
+            f"Block {maxBuBlock} in {a} does not have a spatial grid. Cannot rotate."
+        )
+    maxBuPinLocation = maxBurnupLocator(maxBuBlock)
+    # No need to rotate if max burnup pin is the center
+    if maxBuPinLocation.i == 0 and maxBuPinLocation.j == 0:
+        return 0
+
+    if aPrev is not a:
+        blockAtPreviousLocation = aPrev[a.index(maxBuBlock)]
     else:
-        # transfer percentBuMax rotated pin index to non-rotated pin index
-        if aPrevDetailFlag:
-            # aPrev has pin detail
-            # Determine which of 6 possible rotated pin indices had the lowest power when aPrev was here.
-            prevAssemPowHereMIN = float("inf")
-
-            for possibleRotation in range(6):
-                index = getIndexOfRotatedCell(maxBuPinIndexAssem, possibleRotation)
-                # get pin power at this index in the previously assembly located here
-                # power previously at rotated index
-                prevAssemPowHere = aPrev[bIndexMaxBu].p.linPowByPin[index - 1]
-
-                if prevAssemPowHere is not None:
-                    runLog.debug(
-                        "Previous power in rotation {0} where pinLoc={1} is {2:.4E} W/cm"
-                        "".format(possibleRotation, index, prevAssemPowHere)
-                    )
-                    if prevAssemPowHere < prevAssemPowHereMIN:
-                        prevAssemPowHereMIN = prevAssemPowHere
-                        rot = possibleRotation
-        else:
-            raise ValueError(
-                "Cannot perform detailed rotation analysis without pin-level "
-                "flux information."
-            )
+        blockAtPreviousLocation = maxBuBlock
+
+    previousLocations = blockAtPreviousLocation.getPinLocations()
+    previousPowers = blockAtPreviousLocation.p.linPowByPin
+    if len(previousLocations) != len(previousPowers):
+        raise ValueError(
+            f"Inconsistent pin powers and number of pins in {blockAtPreviousLocation}. "
+            f"Found {len(previousLocations)} locations but {len(previousPowers)} powers."
+        )
 
-        runLog.debug("Best relative rotation is {0}".format(rot))
-        return rot
+    ringPowers = {
+        (loc.i, loc.j): p for loc, p in zip(previousLocations, previousPowers)
+    }
+
+    targetGrid = blockAtPreviousLocation.spatialGrid
+    candidateRotation = 0
+    candidatePower = ringPowers.get((maxBuPinLocation.i, maxBuPinLocation.j), math.inf)
+    runLog.debug(f"::Checking possible rotations for {a}")
+    runLog.debug(
+        f"::rotation={candidateRotation}::power after rotation={candidatePower}::location={maxBuPinLocation}"
+    )
```

Comment:
I'm not sure if I think these are useful enough to leave in the codebase.

Just IMO.

Reply:
I intend to remove them after I've run production-level tests and convinced myself it's still working